litex/soc/integration/builder.py

```diff
@@ -109,9 +109,9 @@ def __init__(self, soc,
         self.build_backend    = build_backend
 
         # Exports.
-        self.csr_csv  = csr_csv
-        self.csr_json = csr_json
-        self.csr_svd  = csr_svd
+        self.csr_csv  = os.path.join(self.output_dir, csr_csv)  if output_dir and csr_csv  else csr_csv
```

Comment:
I like the idea of this patch, but when I tested it, it doesn't seem to work without *explicitly* specifying `--output-dir` as well. The output dir defaults to `./build/<boardname>/` (with further `software/` and `gateware/` subdirectories beneath that) in the absence of an explicit `--output-dir` value.

Am I missing something? If not, would be nice if this worked w/o requiring an explicit output dir specified on the build command line...